### ISSUE TITLE
Broken lint on class property transform after upgrading to v8.1.2

### ISSUE THREAD

**Issue by doodlewind** [babel/babel-eslint#564]
Hi, this underlying library is awesome and fundamental, thanks!

I was using `babel-eslint@8.0.2` together with latest `standard` and `babel-presets-stage-0`. With this config, following code snippet works fine:

``` js
class Foo {
  bar = null
  baz = () => true
}
```

In VSCode, `this.bar` can also be added to automatic completion list. Running `standard` from command line yields no error.

After upgrading this library to `v8.1.2`, `standard` complains:

```
standard: Use JavaScript Standard Style (https://standardjs.com)
  /Users/ewind/code/playground/src/index.js:2:3: 'bar' is not defined.
  /Users/ewind/code/playground/src/index.js:3:3: 'baz' is not defined.
```

Neither can VSCode "sense" `this.bar` then. Am I missing some extra configuration?

**Comment by Skylsmoi** [comment]
I have the same issue, using standard loader too.

I can add that after testing every versions, the issue starts at version 8.1.0

**Comment by doodlewind** [comment]
I investigate a bit and found that a patch for scope analysis is introduced in #542, which is [not compatible](https://github.com/babel/babel-eslint/pull/542#issuecomment-353943499) with `babel-eslint@7.x`. Since `standard@10.x` uses `babel-eslint@7.x`, this patch may introduce unexpected behaviors for `standard@10.x`.

The good news is that in `standard@11.x`, dependency version is updated to `babel-eslint@8.x`. I've tested several permutation and find two workarounds for now:

1. Install standard@10.x with babel-eslint@7.x
2. Install standard@11.x with babel-eslint@8.x

Guess this is not an issue about `babel-eslint`, closing for now.

**Comment by pex** [comment]
Unfortunately I am having the exact same issue using [`XO`](https://github.com/sindresorhus/xo) and `babel-eslint` version 8.1.2. Here's a repo reproducing the case: https://github.com/pex/xo-babel-eslint-class-properties

Everything works fine using `babel-eslint` version 8.0.3.

**Comment by RKushnir** [comment]
@doodlewind @ematipico

> Guess this is not an issue about babel-eslint, closing for now.

> Closing the issue as it's not a problem of standard but of babel-eslint@8.1.3 (https://github.com/standard/standard/issues/1035)

Please, leave at least one of the issues open, so it doesn't get lost.

**Comment by RKushnir** [comment]
Looks like the issue has been resolved. Works for me on 8.2.1 with the standard 11.x.
